### DISCUSSION TITLE
To fetch socket programmatically in Uvicorn.

### DISCUSSION THREAD

**Discussion by anuragsrivstv** [Kludex/uvicorn#1703] [Q&A] [answered]
Hi all, I'm doing a migration of a codebase from Gunicorn to Uvicorn , mainly because of the fact that Uvicorn is windows compatible.

There are some unit test scenario in which we want to close the socket explicitly if an exception occurs while invoking a get request(Flask server).  We were able to do it in gunicorn as it has gunicorn.socket in request environment variable. Is there any way in Uvicorn to fetch the socket and close it explicitly ?

**Comment by xavatu** [comment]
Hey, I'm interested in this too.

**Comment by Kludex** [comment, marked as answer]
https://github.com/encode/uvicorn/issues/761#issuecomment-1287679527
